### PR TITLE
[GPU] Support 7D and 8D tensors

### PR DESCRIPTION
**Pull request by vladimir-paramuzov**
### Details:
 - Increased max supported tensor rank to 8d
 - Enable new shape inference for >6d tensors by default
 - Support up to 8d tensors in reshape, eltwise, reduce and permute primitives

### Tickets:
 - *100599, 101932*

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/include/fetch_utils.cl [hunk 1]
```diff
@@ -4,26 +4,42 @@
...
+    return INPUT0_GET_INDEX(b, f, v, u, w, z, y, x);
 #else
 #error [GPU] Unsupported input tensor rank in get_input_index function
 #endif
 }
 
-inline uint FUNC(get_output_index)(OPTIONAL_SHAPE_INFO_ARG uint b, uint f, uint w, uint z, uint y, uint x) {
+inline uint FUNC(get_input_index)(OPTIONAL_SHAPE_INFO_ARG uint b, uint f, uint w, uint z, uint y, uint x) __attribute__((overloadable)) {
+    return FUNC_CALL(get_input_index)(OPTIONAL_SHAPE_INFO_TENSOR b, f, 0, 0, w, z, y, x);
+}
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/include/fetch_utils.cl:25]: Just out of curiosity: wouldn't this cause registers pressure increase? And wouldn't prevent function inlining?..

File: src/plugins/intel_gpu/src/kernel_selector/jitter.cpp [hunk 2]
```diff
@@ -537,6 +548,53 @@ JitDefinitions DataTensorJitConstant::GetDefinitions() const {
...
+        if (is_common_8d_layout) {
+            index_func_name = _name + "_GET_INDEX(b, f, v, u, w, z, y, x)";
+            safe_index_func_name = _name + "_GET_INDEX_SAFE(b, f, v, u, w, z, y, x)";
+            raw_index_func_name = _name + "_GET_INDEX_RAW(b, f, v, u, w, z, y, x)";
+
+            index_func_val = "GET_DATA_INDEX_8D(" + _name + ", b, f, v, u, w, z, y, x)";
+            safe_index_func_val = "GET_DATA_INDEX_8D_SAFE(" + _name + ", b, f, v, u, w, z, y, x)";
+            raw_index_func_val = "GET_DATA_INDEX_8D_RAW(" + _name + ", b, f, v, u, w, z, y, x)";
+        } else {
+            // TODO: implement support of non-default layouts with 6 channels
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/kernel_selector/jitter.cpp:595]: `OPENVINO_ASSERT(is_common_8d_layout, ...)`

File: src/plugins/intel_gpu/src/runtime/layout.cpp [hunk 3]
```diff
@@ -518,97 +518,45 @@ ov::PartialShape layout::transform(const ov::PartialShape& pshape, cldnn::format
         return pshape;
     }
 
-    cldnn::tensor::value_type default_size = -1;
+    int32_t default_size = -1;
     auto shape = pshape.to_shape();
-    std::vector<tensor::value_type> dims;
+    std::vector<int32_t> dims;
     for (auto dim : shape) {
         dims.push_back(static_cast<tensor::value_type>(dim));
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/runtime/layout.cpp:525]:
```suggestion
        dims.push_back(static_cast<int32_t>(dim));
```

File: src/plugins/intel_gpu/include/intel_gpu/runtime/tensor.hpp [hunk 4]
```diff
@@ -525,17 +524,33 @@ struct tensor {
                 continue;
             }
 
+            if (new_fmt != format::bfuwzyx && c == 'u') {
```
[the remark this answers is not in the document]
Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/include/intel_gpu/runtime/tensor.hpp]: Updated this impl and added tests

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 5]
```diff
@@ -1582,8 +1586,17 @@ format layout_optimizer::get_preferred_format(program_node& node) {
...
             auto in_lay_rank = node.get_dependency(i).get_output_layout(false).get_rank();
-            if (in_lay_rank != out_lay_rank)
-                node.set_preferred_input_fmt(i, get_preferred_format(node.get_dependency(i)));
+            if (in_lay_rank != out_lay_rank) {
+                auto fmt = get_preferred_format(node.get_dependency(i));
+                // Check if selected format can be adjusted to the required output rank
+                // If no, use default fotmat instead
+                try {
+                    format::adjust_to_rank(fmt, out_lay_rank);
+                } catch (std::exception&) {
```
[the remark this answers is not in the document]
Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp]: Updated

**Pull request merged**